src/main/kotlin/com/openlattice/data/storage/PostgresEntityDataQueryService.kt

```diff
@@ -282,12 +296,21 @@ class PostgresEntityDataQueryService(
                                 )
                             }
                     )
-                    StatementHolder(connection, statement, rs)
+
+                    val cleanupStatement = connection.createStatement()
+                    val cleanupResultSet = cleanupStatement
+                            .executeQuery("DELETE FROM ${QUERIES.name} WHERE ${QUERY_ID.name} = $queryId")
+
+                    StatementHolder(connection, statement, rs, listOf(cleanupStatement), listOf(cleanupResultSet))
                 },
                 adapter
         )
     }
 
+    fun createQueryTable(entityKeyIds: Map<UUID, Optional<Set<UUID>>>) {
+
```

Comment:
Magic happens here?

Reply:
Ended up going in a different direction :)